File: light_switch_sctunit/src-gen/org/yakindu/scr/light_switch/Light_SwitchStatemachine.java
package org.yakindu.scr.light_switch;

public class Light_SwitchStatemachine implements ILight_SwitchStatemachine {

	protected class SCInterfaceImpl implements SCInterface {
	
		private boolean operate;
		
		public void raiseOperate() {
			operate = true;
		}
		
		protected void clearEvents() {
			operate = false;
		}
	}
	
	protected SCInterfaceImpl sCInterface;
	
	private boolean initialized = false;
	
	public enum State {
		main_region_Off,
		main_region_On,
		$NullState$
	};
	
	private final State[] stateVector = new State[1];
	
	private int nextStateIndex;
	
	public Light_SwitchStatemachine() {
		sCInterface = new SCInterfaceImpl();
	}
	
	public void init() {
		this.initialized = true;
		for (int i = 0; i < 1; i++) {
			stateVector[i] = State.$NullState$;
		}
		clearEvents();
		clearOutEvents();
	}
	
	public void enter() {
		if (!initialized) {
			throw new IllegalStateException(
					"The state machine needs to be initialized first by calling the init() function.");
		}
		enterSequence_main_region_default();
	}
	
	public void exit() {
		exitSequence_main_region();
	}
	
	/**
	 * @see IStatemachine#isActive()
	 */
	public boolean isActive() {
		return stateVector[0] != State.$NullState$;
	}
	
	/** 
	* Always returns 'false' since this state machine can never become final.
	*
	* @see IStatemachine#isFinal()
	*/
	public boolean isFinal() {
		return false;
	}
	/**
	* This method resets the incoming events (time events included).
	*/
	protected void clearEvents() {
		sCInterface.clearEvents();
	}
	
	/**
	* This method resets the outgoing events.
	*/
	protected void clearOutEvents() {
	}
	
	/**
	* Returns true if the given state is currently active otherwise false.
	*/
	public boolean isStateActive(State state) {
	
		switch (state) {
		case main_region_Off:
			return stateVector[0] == State.main_region_Off;
		case main_region_On:
			return stateVector[0] == State.main_region_On;
		default:
			return false;
		}
	}
	
	public SCInterface getSCInterface() {
		return sCInterface;
	}
	
	public void raiseOperate() {
		sCInterface.raiseOperate();
	}
	
	private boolean check_main_region_Off_tr0_tr0() {
		return sCInterface.operate;
	}
	
	private boolean check_main_region_On_tr0_tr0() {
		return sCInterface.operate;
	}
	
	private void effect_main_region_Off_tr0() {
		exitSequence_main_region_Off();
		enterSequence_main_region_On_default();
	}
	
	private void effect_main_region_On_tr0() {
		exitSequence_main_region_On();
		enterSequence_main_region_Off_default();
	}
	
	/* 'default' enter sequence for state Off */
	private void enterSequence_main_region_Off_default() {
		nextStateIndex = 0;
		stateVector[0] = State.main_region_Off;
	}
	
	/* 'default' enter sequence for state On */
	private void enterSequence_main_region_On_default() {
		nextStateIndex = 0;
		stateVector[0] = State.main_region_On;
	}
	
	/* 'default' enter sequence for region main region */
	private void enterSequence_main_region_default() {
		react_main_region__entry_Default();
	}
	
	/* Default exit sequence for state Off */
	private void exitSequence_main_region_Off() {
		nextStateIndex = 0;
		stateVector[0] = State.$NullState$;
	}
	
	/* Default exit sequence for state On */
	private void exitSequence_main_region_On() {
		nextStateIndex = 0;
		stateVector[0] = State.$NullState$;
	}
	
	/* Default exit sequence for region main region */
	private void exitSequence_main_region() {
		switch (stateVector[0]) {
		case main_region_Off:
			exitSequence_main_region_Off();
			break;
		case main_region_On:
			exitSequence_main_region_On();
			break;
		default:
			break;
		}
	}
	
	/* The reactions of state Off. */
	private void react_main_region_Off() {
		if (check_main_region_Off_tr0_tr0()) {
			effect_main_region_Off_tr0();
		}
	}
	
	/* The reactions of state On. */
	private void react_main_region_On() {
		if (check_main_region_On_tr0_tr0()) {
			effect_main_region_On_tr0();
		}
	}
	
	/* Default react sequence for initial entry  */
	private void react_main_region__entry_Default() {
		enterSequence_main_region_Off_default();
	}
	
	public void runCycle() {
		if (!initialized)
			throw new IllegalStateException(
					"The state machine needs to be initialized first by calling the init() function.");
		clearOutEvents();
		for (nextStateIndex = 0; nextStateIndex < stateVector.length; nextStateIndex++) {
			switch (stateVector[nextStateIndex]) {
			case main_region_Off:
				react_main_region_Off();
				break;
			case main_region_On:
				react_main_region_On();
				break;
			default:
				// $NullState$
			}
		}
		clearEvents();
	}
}

File: org.yakindu.sct.examples.codegen.python/src-gen/LightSwitch/lightswitch/lightswitch_statemachine.py
# Implementation of statechart lightswitch_statemachine.

import warnings
# implemented interfaces:
from .lightswitch_statemachine_interfaces import SCI_user
# to store states:
from enum import Enum

class LightSwitch:

	"""
	
	Implementation of the state machine 'LightSwitch'.
	
	"""
	
	def __init__(self):
		""" Declares all necessary variables including list of states, histories etc. 
		"""
		self.sci_user = SCI_user(self)
		self.initialized = False
		self.state_vector = [None] * 1		
		self.next_state_index = None		
		# enumeration of all states:
		self.State = Enum('State', '\
		main_region_off,\
		main_region_on,\
		null_state')
	
	def init(self):
		"""	Initializes the state machine by checking the timer, 
		initializing states and clearing events.
		"""
		self.initialized = True
		for state_index in range(1):
			self.state_vector[state_index] = self.State.null_state
		self._clear_events()
		self.sci_user.brightness = 0
	
	def enter(self):
		if self.initialized is not True:
			raise ValueError(
					'The state machine needs to be initialized first by calling the init() function.')
		self.enter_sequence_main_region_default()
	
	def exit(self):
		"""	Exit the the state machine.
		"""
		self.exit_sequence_main_region()
	
	def is_active(self):
		""" @see IStatemachine#is_active()
		"""
		return (self.state_vector[0] is not self.State.null_state)
	
	def is_final(self):
		"""Always returns 'false' since this state machine can never become final.
		@see IStatemachine#is_final()
		"""
		return False
			
	def _clear_events(self):
		""" Resets incoming events (time events included).
		"""
		self.sci_user.clear_events()
	
	def is_state_active(self, state):
		""" Returns True if the given state is currently active otherwise false.
		"""
		s = state.name
		if s == 'main_region_off':
			return self.state_vector[0] == self.State.main_region_off
		elif s == 'main_region_on':
			return self.state_vector[0] == self.State.main_region_on
		else:
			return False
	
	def entry_action_main_region_off(self):
		self.sci_user.brightness = 0
		
	def enter_sequence_main_region_off_default(self):
		self.entry_action_main_region_off()
		self.next_state_index = 0
		self.state_vector[0] = self.State.main_region_off
		
	def enter_sequence_main_region_on_default(self):
		self.next_state_index = 0
		self.state_vector[0] = self.State.main_region_on
		
	def enter_sequence_main_region_default(self):
		self.react_main_region__entry_default()
		
	def exit_sequence_main_region_off(self):
		self.next_state_index = 0
		self.state_vector[0] = self.State.null_state
		
	def exit_sequence_main_region_on(self):
		self.next_state_index = 0
		self.state_vector[0] = self.State.null_state
		
	def exit_sequence_main_region(self):
		state = self.state_vector[0].name
		if state == 'main_region_off':
			self.exit_sequence_main_region_off()		
		elif state == 'main_region_on':
			self.exit_sequence_main_region_on()
		
	def react_main_region__entry_default(self):
		self.enter_sequence_main_region_off_default()
		
	def react(self):
		return False
	
	
	def main_region_off_react(self,  try_transition):
		did_transition = try_transition
		
		if try_transition:
			if (self.react()) == False:
				if self.sci_user.on_button:
					self.exit_sequence_main_region_off()
					self.sci_user.brightness = 1
					self.enter_sequence_main_region_on_default()
				else:
					did_transition = False
		
		return did_transition
	
	
	def main_region_on_react(self,  try_transition):
		did_transition = try_transition
		
		if try_transition:
			if (self.react()) == False:
				if self.sci_user.off_button:
					self.exit_sequence_main_region_on()
					self.enter_sequence_main_region_off_default()
				elif ((self.sci_user.on_button) and ((self.sci_user.brightness) < 10)):
					self.exit_sequence_main_region_on()
					self.sci_user.brightness = self.sci_user.brightness + 1
					self.enter_sequence_main_region_on_default()
				else:
					did_transition = False
		
		return did_transition
	
	
	def run_cycle(self):
		""" Starts a cycle in the state machine. 
		"""
		if self.initialized is not True:
			raise ValueError(
					'The state machine needs to be initialized first by calling the init() function.')
		self.next_state_index = 0
		while self.next_state_index < len(self.state_vector):
			if self.state_vector[self.next_state_index].name == 'main_region_off' :
				self.main_region_off_react(True)
			elif self.state_vector[self.next_state_index].name == 'main_region_on' :
				self.main_region_on_react(True)
			self.next_state_index += 1
			
		self._clear_events()
